src/Components/Components/src/RenderHandle.cs

```diff
@@ -2,13 +2,12 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Threading.Tasks;
 using Microsoft.AspNetCore.Components.Rendering;
 
 namespace Microsoft.AspNetCore.Components
 {
     /// <summary>
-    /// Allows a component to notify the renderer that it should be rendered.
+    /// Allows a component to interact with its renderer.
```

Comment:
I'm liking what `RenderHandle` does the overall design of the system, it can act as a firewall between components and the rendering system. This is an example of how we can expose a renderer concept (threading/dispatching) to components, without exposing the renderer.

Reply:
Yep! Sounds good. The original intent for `RenderHandle` was what you mention about firewalling the rendering system from components, plus a way to ensure that components can only render themselves and not other components. That is, components don't get told what their ID is, nor can they request renders for other arbitrary components by ID or by object reference. This is so you can always reason about when a particular component can be rendered (i.e., it's when that component itself asks to be rendered, and only then), but you're still free to open up access more broadly, e.g., by adding a public "render me" method on your component.